### PR TITLE
Replace cachecontrol package with joblib.Memory

### PR DESCRIPTION
**Pull request by kernc**
Likely installed, maintained by sklearn guys, assumed to work well.

### CODE REVIEW

File: orangecontrib/imageanalytics/image_embedder.py [hunk 1]
```diff
@@ -56,11 +56,13 @@ def __init__(self, model, layer, server_url='api.biolab.si:8080'):
...
         self._cache_dict = self._init_cache()
-        self._session = cachecontrol.CacheControl(
-            requests.session(),
-            cache=cachecontrol.caches.FileCache(
-                join(cache_dir(), __name__ + ".ImageEmbedder.httpcache"))
-        )
+
+        cached = Memory(
+            cachedir=join(cache_dir(), __name__ + ".ImageEmbedder.httpcache"),
+            compress=1, bytes_limit=100e6).cache
```
Comment (ales-erjavec) [inline orangecontrib/imageanalytics/image_embedder.py]: I think you still have to call `Memory.reduce_size` manually for the bytes_limit to have any effect (at least per joblib's [changelog](https://github.com/joblib/joblib/blob/master/CHANGES.rst).

File: orangecontrib/imageanalytics/image_embedder.py [hunk 2]
```diff
@@ -229,7 +231,7 @@ def _load_image_from_url_or_local_path(self, file_path):
         urlparts = urlparse(file_path)
         if urlparts.scheme in ('http', 'https'):
             try:
-                file = self._session.get(file_path, stream=True).raw
+                file = BytesIO(self._requests_get_bytes(file_path, stream=True))
```
Comment (ales-erjavec) [inline orangecontrib/imageanalytics/image_embedder.py]: No need for `stream=True`. Its just confusing now.

**Comment by ales-erjavec** [comment]
> Likely installed, maintained by sklearn guys, assumed to work well.

... but is not a HTTP cache. I don't really care one way or the other. Go for it if you think this is better.

File: orangecontrib/imageanalytics/image_embedder.py (hunk 1 above)
Comment (kernc) [inline orangecontrib/imageanalytics/image_embedder.py]: Thanks.

**Comment by kernc** [comment]
Is this a setback? Indeed it's a much dumber _callable_ cache, here transparently mapping URLs to bytes. It has a simple API and comes preinstalled, avoiding three dependencies, one of which advertised as abandoned. I read this as slightly better, but certainly I'm proven to be naive. Just proposing an alternative. 😇

**Comment by jerneju** [comment]
@kernc : What is the status of this PR?

**Comment by kernc** [comment]
PoC, stale.

**Pull request closed without merging**